lib/runtime-abi/src/vfs/device_file.rs

```diff
@@ -0,0 +1,84 @@
+use crate::vfs::file_like::{FileLike, Metadata};
+use failure::Error;
+use std::io;
+use std::io::{Seek, Write};
+
+pub struct Stdin;
+pub struct Stdout;
+pub struct Stderr;
+
+impl FileLike for Stdin {
+    fn metadata(&self) -> Result<Metadata, Error> {
+        unimplemented!()
+    }
+
+    fn write_file(&mut self, _buf: &[u8], _offset: usize) -> Result<usize, io::Error> {
+        unimplemented!()
+    }
+
+    fn read_file(&mut self, _buf: &mut [u8], _offset: usize) -> Result<usize, io::Error> {
```

Comment:
Reading isn't implemented for stdin?

Reply:
Reading from stdin was not needed to get bundled nginx working. Seems like something we can kick the can on. Would you prefer we do it now?

Reply:
Nah, we don't need it now.